Add documentation for Farms (#44)
Leaf-Agriculture/docs: docs/field_boundaries_endpoints.md

DELETE /users/{leafUserId}/fields/{id}
```

#### Farm Resource

> Form of a Farm:
> ```json

Reviewer: I think that for reasons of markdown syntax it's better or necessary to have one blank line here
```suggestion
Form of a Farm:

```json
```

{
  "id": "long",
  "name": "str",

>   "providerName": "str",

Reviewer 2: we have to define if the convention will be "providerName" or "provider"... we have both now

Reviewer: Yep. I vote for `provider`

>   "providerFarmId": "UUID",

Reviewer 2: can we remove this from the response? It's unnecessary, from my perspective....

Reviewer 2: I am referring to "providerFarmId", this is the id of the farm on the provider right?

Reviewer: Yes, it is. Why unnecessary? datasync can create farms, so they will have this attribute set

  const TOKEN = 'YOUR_TOKEN'

  const endpoint ='https://api.withleaf.io/services/fields/api/fields'

>   const headers = { 'Authorization': `Bearer ${TOKEN}` }

Reviewer 3: Why have you changed this? It is valid JavaScript

Reviewer: Approve it

}
```

### `GET /farms`
Gets a paged list of all Farms. It is possible to pass some query parameters.

- `page`, an integer specifying the page being fetched
- `size`, an integer specifying the size of the page

The parameters are used exclusively for paging through results.

#### Response
A JSON array containing Farms.

<Tabs
  defaultValue="sh"
  values={[
    { label: 'cURL', value: 'sh', },
    { label: 'Python', value: 'py', },
    { label: 'JavaScript', value: 'js', },
    { label: 'JSON Response', value: 'res', },
  ]
}>
  <TabItem value="js">

  ```js
  const axios = require('axios')
  const TOKEN = 'YOUR_TOKEN'

  const endpoint ='https://api.withleaf.io/services/fields/api/farms'

>   const headers = { 'Authorization': `Bearer ${TOKEN}` }

Reviewer: ```suggestion
  const headers = { 'Authorization': 'Bearer ${TOKEN}' }
```

Reviewer: Same in other places

  const axios = require('axios')
  const TOKEN = 'YOUR_TOKEN'

  const endpoint = 'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
  const headers = { 'Authorization': `Bearer ${TOKEN}` }

  axios.get(endpoint, { headers })
      .then(res => console.log(res.data))
      .catch(console.error)
  ```

  </TabItem>
  <TabItem value="py">

  ```py
  import requests

  TOKEN = 'YOUR_TOKEN'

  endpoint = 'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
  headers = {'Authorization': f'Bearer {TOKEN}'}

  response = requests.get(endpoint, headers=headers)
  print(response.json())
  ```

  </TabItem>
  <TabItem value="sh">

  ```shell
  curl -X GET \
      -H 'Authorization: Bearer YOUR_TOKEN' \
      'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
  ```

  </TabItem>
  <TabItem value="res">

  ```json
  {
    "id": 1551010,
    "name": "name",
    "providerName": "JohnDeere",
    "providerFarmId": "00000000-0000-0000-0000-000000000000",
    "leafUserId": "00000000-0000-0000-0000-000000000000",
    "fieldIds": ["00000000-0000-0000-0000-000000000000"],
    "growerId": 123
  }
  ```

  </TabItem>
</Tabs>

> ### `DELETE users/{leafUserId}/farms/{id}`

Reviewer: "Only at leaft" for the 1st question. "Maybe" for the 2nd. If that farm was created from pooling then yes, we do will create it again. We know 0 real use case for datasync, so I think documenting this kind of endpoint are 100% a product decision. The fact is that all endpoint that allows writting are conflicting with datasync. Currently we're simply ignoring this fact.

Deletes the farm with the given id.

<Tabs
  defaultValue="sh"
  values={[
    { label: 'cURL', value: 'sh', },
    { label: 'Python', value: 'py', },
    { label: 'JavaScript', value: 'js', },
  ]
}>
  <TabItem value="js">

  ```js
  const axios = require('axios')
  const TOKEN = 'YOUR_TOKEN'

  const endpoint ='https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
  const headers = { 'Authorization': `Bearer ${TOKEN}` }

  axios.delete(endpoint, { headers })
      .then(res => console.log(res.data))
      .catch(console.error)
  ```

  </TabItem>
  <TabItem value="py">

  ```py
  import requests

  TOKEN = 'YOUR_TOKEN'

  endpoint = 'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
  headers = {'Authorization': f'Bearer {TOKEN}'}

  response = requests.delete(endpoint, headers=headers)
  print(response.json())
  ```

  </TabItem>
  <TabItem value="sh">

  ```shell
  curl -X DELETE \
      -H 'Authorization: Bearer YOUR_TOKEN' \
      'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
  ```

  </TabItem>
</Tabs>

> ### `POST users/{leafUserId}/farms`

Reviewer: Nope, although we can add a farm to a field on field creation. Does this functionality you're asking about make sense?

> Creates a Farm. The request's body can provide values for the farm

Reviewer: ```suggestion
Creates a Farm. The request's body provides values for the farm
```

    { label: 'cURL', value: 'sh', },
    { label: 'Python', value: 'py', },
    { label: 'JavaScript', value: 'js', },
  ]
}>
  <TabItem value="js">

  ```js
  const axios = require('axios')
  const TOKEN = 'YOUR_TOKEN'

  const endpoint ='https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms'
  const headers = { 'Authorization': `Bearer ${TOKEN}` }

  const data = {
    name: "name",
    growerId: 12345
  }

  axios.post(endpoint, { headers, data })
      .then(res => console.log(res.data))
      .catch(console.error)
  ```

  </TabItem>
  <TabItem value="py">

  ```py
  import requests

  TOKEN = 'YOUR_TOKEN'

  endpoint = 'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms'
  headers = {'Authorization': f'Bearer {TOKEN}'}

  data = {
    'name': "name",
    'growerId': 12345
  }

  response = requests.post(endpoint, headers=headers, json=data)
  print(response.json())
  ```

  </TabItem>
  <TabItem value="sh">

  ```shell
  curl -X POST \
      -H 'Authorization: Bearer YOUR_TOKEN' \
      -d '{ "name": "name", "growerId": 12345 }'
      'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms'
  ```

  </TabItem>
</Tabs>

> ### `PUT users/{leafUserId}/farms`

Reviewer: Nope
